src/ai/llm/services/factory.ts

```
@@ -84,29 +90,56 @@ function _createInBuiltLLMService(
 }
 
-function _createVercelModel(provider: string, model: string): LanguageModelV1 {
+function _createVercelModel(llmConfig: LLMConfig): LanguageModelV1 {
+    const provider = llmConfig.provider;
+    const model = llmConfig.model;
+    const apiKey = extractApiKey(llmConfig);
+
     switch (provider.toLowerCase()) {
         case 'openai':
-            return openai(model);
+            const baseURL = getOpenAICompatibleBaseURL(llmConfig);
```

Biome (1.9.4)

[error] Other switch clauses can erroneously access this declaration.
Wrap the declaration in a block to restrict its access to the switch clause.

The declaration is defined in this switch clause:

Unsafe fix: Wrap the declaration in a block.


(lint/correctness/noSwitchDeclarations)